consensus/types/height_vote_set.go

```diff
@@ -158,6 +158,20 @@ func (hvs *HeightVoteSet) POLInfo() (polRound int, polBlockID types.BlockID) {
 	return -1, types.BlockID{}
 }
 
+// For Make proposal
+func (hvs *HeightVoteSet) POLInfoExcludeCurrentRound() (polRound int, polBlockID types.BlockID) {
+	hvs.mtx.Lock()
+	defer hvs.mtx.Unlock()
+	for r := hvs.round - 1; r >= 0; r-- {
+		rvs := hvs.getVoteSet(r, types.PrevoteType)
+		polBlockID, ok := rvs.TwoThirdsMajority()
+		if ok && len(polBlockID.Hash) != 0 {
```

Comment:
Should we do the same thing for `POLInfo`? Can we extract common code into a private function?

```
func (hvs *HeightVoteSet) POLInfo() (polRound int, polBlockID types.BlockID) {
	return polInfo(hvs.round)
}

func (hvs *HeightVoteSet) POLInfoExcludingCurrentRound() (polRound int, polBlockID types.BlockID) {
        if hvs.round == 0 {
            return -1, types.BlockID{}
        }
	return polInfo(hvs.round-1)
}

func (hvs *HeightVoteSet) polInfo(round int) (polRound int, polBlockID types.BlockID) {
hvs.mtx.Lock()
	defer hvs.mtx.Unlock()
	for r := hvs.round; r >= 0; r-- {
		rvs := hvs.getVoteSet(r, types.PrevoteType)
		polBlockID, ok := rvs.TwoThirdsMajority()
		if ok {
			return r, polBlockID
		}
	}
	return -1, types.BlockID{}
}
```

Reply:
len(polBlockID.Hash) > 0 could not be added , because enterPrecommit will call POLInfo too. It's possible that 2/3+ prevote nil  caused to enterPrecommit.